x-pack/plugins/security_solution/server/endpoint/services/actions/clients/endpoint_actions_client.ts

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { HapiReadableStream } from '../../../../types';
+import type { ResponseActionsApiCommandNames } from '../../../../../common/endpoint/service/response_actions/constants';
+import { updateCases } from '../create/update_cases';
+import type { CreateActionPayload } from '../create/types';
+import type {
+  ExecuteActionRequestBody,
+  GetProcessesRequestBody,
+  IsolationRouteRequestBody,
+  ResponseActionGetFileRequestBody,
+  UploadActionApiRequestBody,
+  ResponseActionsRequestBody,
+} from '../../../../../common/api/endpoint';
+import { BaseResponseActionsClient } from '../../../lib/response_actions/base_actions_provider';
+import type {
+  ActionDetails,
+  HostMetadata,
+  GetProcessesActionOutputContent,
+  KillOrSuspendProcessRequestBody,
+  KillProcessActionOutputContent,
+  ResponseActionExecuteOutputContent,
+  ResponseActionGetFileOutputContent,
+  ResponseActionGetFileParameters,
+  ResponseActionParametersWithPidOrEntityId,
+  ResponseActionsExecuteParameters,
+  ResponseActionUploadOutputContent,
+  ResponseActionUploadParameters,
+  SuspendProcessActionOutputContent,
+  HostMetadataInterface,
+  ImmutableObject,
+} from '../../../../../common/endpoint/types';
+
+export class EndpointActionsClient extends BaseResponseActionsClient {
```

Comment:
FYI: testing of this client is covered by existing tests for both the Route handler for response actions and the individual service functions under `services/actions/*`